src/textual/binding.py

```diff
@@ -67,6 +69,16 @@ def make_bindings(bindings: Iterable[BindingType]) -> Iterable[Binding]:
                         )
                     binding = Binding(*binding)
 
+                binding_keys = binding.key.split(",")
+                for key in binding_keys:
+                    # normalized keys have modifier characters (like '+') replaced with Python-identifier substitutes
+                    if not _normalize_key(key).strip().isidentifier():
+                        msg = f"BINDINGS key '{key}' is invalid"
+                        suggested_key = _get_suggested_binding_key(key.strip())
+                        if suggested_key:
+                            msg += f"; try replacing it with '{suggested_key}'"
+                        raise BindingError(msg)
+
```

Comment:
I'm wondering: rather than loop over the keys twice, first to validate and then to actually create the bindings; perhaps do the validation within the binding loop?

Reply:
Tricky to say - in generator functions I like to avoid too many sychronization blocks like this, but.. validating state before yielding any/partial output seems like a good idea too, generally.

I'd probably lean towards a cautious approach to begin with (perform the validation separately, and not worry too much about performance/conciseness yet - as long as this isn't a performance bottleneck).  Open to other considerations, though.